lib/workos/user_management.rb

```diff
@@ -503,19 +503,28 @@ def authenticate_user_with_totp(code:, client_id:, pending_authentication_token:
       # Enroll a user into an authentication factor.
       #
       # @param [String] user_id The id for the user.
+      # @param [String] type The type of the factor to enroll. Only option available is totp.
+      # @param [String] totp_issuer For totp factors. Typically your application
+      #  or company name, this helps users distinguish between factors in authenticator apps.
+      # @param [String] totp_user For totp factors. Used as the account name in authenticator apps.
       #
       # @return WorkOS::AuthenticationFactorAndChallenge
       sig do
         params(
           user_id: String,
+          type: String,
```

Comment:
one idea: perhaps introduce an Enum (something like `FactorType` or `AuthFactorType`) which currently has a single value (`'totp'`)?

Reply:
I've introduced an enum, inside of the User Management module scope, instead of following that pattern of creating as in the global Types namespace. 

Pretty much the same thing that we did for providers, and we're now validating the auth factor: https://github.com/workos/workos-ruby/pull/240/commits/bad936333c8c048f784a1ea652323a628a5cce24

Then let's keep that in mind if one day we add new auth factors, that enum should be updated as well.